src/h5web/providers/mock/metadata-utils.ts

```diff
@@ -138,19 +129,19 @@ export function makeDataset<S extends HDF5Shape, T extends HDF5Type>(
   name: string,
   shape: S,
   type: T,
-  opts: EntityOpts = {}
-): Dataset<S, T> {
-  const { id = name, attributes = [], rawLink } = opts;
+  opts: DatasetOpts = {}
+): MockDataset<S, T> {
+  const { attributes = [], valueId = name, rawLink } = opts;
 
   return {
     uid: nanoid(),
-    id,
     name,
     path: `/${name}`,
     kind: EntityKind.Dataset,
     attributes,
     shape,
     type,
+    value: mockValues[valueId as MockValueId],
```

Comment:
We read the dataset's value from `mockValues` on creation, based on `valueId`. If a `valueId` is not provided, the dataset's `name` is used.